wandb/sdk/wandb_run.py

```diff
@@ -2929,20 +2930,19 @@ def _assert_can_log_artifact(self, artifact) -> None:  # type: ignore
 
     def _prepare_artifact(
         self,
-        artifact_or_path: Union[wandb_artifacts.Artifact, str],
+        artifact_or_path: Union[wandb_artifacts.Artifact, StrPath],
         name: Optional[str] = None,
         type: Optional[str] = None,
         aliases: Optional[List[str]] = None,
     ) -> Tuple[wandb_artifacts.Artifact, List[str]]:
-        if isinstance(artifact_or_path, str):
-            if name is None:
-                name = f"run-{self._run_id}-{os.path.basename(artifact_or_path)}"
-            artifact = wandb.Artifact(name, type)
+        if isinstance(artifact_or_path, (str, os.PathLike)):
+            name = name or f"run-{self._run_id}-{os.path.basename(artifact_or_path)}"
+            artifact = wandb.Artifact(name, type or "dataset")
```

Comment:
I would say, let's keep the change minimal here. What we're trying to achieve is already taken care of, by doing `type: str = "dataset",` here. adding `type: str = "dataset",` in class Artifact(ArtifactInterface) seems redundant, and not necessary for this change.

In fact, model objects (such as `Artifact`), are usually used by the same codebase. Only the api classes (such as this one), are supposed to be used by the users of this library. Since Artifact is useful mostly for internal purpose, I would hope for explicit arguments there, and only allow flexibility on this layer.

Reply:
Agreed, users shouldn't be able to call `wandb.Artifact(name)` this should only be for the shortcut methods `run.log_artifact("....")`